bgpd/bgp_zebra.c

```diff
@@ -1338,12 +1338,14 @@ int bgp_redistribute_set(struct bgp *bgp, afi_t afi, int type, u_short instance)
 		vrf_bitmap_set(zclient->redist[afi][type], bgp->vrf_id);
 	}
 
-	/* Don't try to register if we're not connected to Zebra or Zebra
-	 * doesn't
-	 * know of this instance.
+	/*
+	 * Don't try to register if we're not connected to Zebra or Zebra
+	 * doesn't know of this instance.
+	 *
+	 * When we come up later well resend if needed.
 	 */
 	if (!bgp_install_info_to_zebra(bgp))
-		return CMD_WARNING_CONFIG_FAILED;
+		return CMD_SUCCESS;
```

Comment:
So this will remove the user notification that the command may not have any effect, right?  Are you okay with this?

Reply:
it will have an effect once the instance is turned on as that we install it then.

Reply:
I understood that, but a warning may remind the user to turn on the instance, no?

Reply:
So we have to turn on the instance to even get to whether or not we are going to do redistribution or not.  So I do not understand your point here.

We have done this:
router bgp 523
   redistribute static
!

bgp is coming up, but has not fully made the connection to zebra due to a non-integrated config bring up.

Reply:
it's fine, but it is a change to a cli user typing the command.